cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/cache/DynamicDatasetCacheTest.java

```diff
@@ -121,10 +128,17 @@ protected void testDatasetCache(@Nullable Map<String, TestDataset> datasetMap)
     TestDataset b1 = cache.getDataset("b", B_ARGUMENTS, AccessType.READ);
     TestDataset b2 = cache.getDataset("b", B_ARGUMENTS, AccessType.WRITE);
     Assert.assertSame(b, b1);
-
     // assert that b1 and b2 are the same, even though their accessType is different
     Assert.assertSame(b1, b2);
 
+    // Test this for cross namespace access
+    TestDataset a_n2 = cache.getDataset(NAMESPACE2.getId(), "a2");
+    TestDataset a1_n2 = cache.getDataset(NAMESPACE2.getId(), "a2");
+    TestDataset a2_n2 = cache.getDataset(NAMESPACE2.getId(), "a2");
```

Comment:
also makes sense to validate that this instance is not the same as the instance of a from the other namespace.